xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/internal/AttachmentsListener.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.internal;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.observation.EventListener;
+import org.xwiki.observation.event.Event;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.internal.event.AttachmentAddedEvent;
+
+import static java.util.Collections.singletonList;
+import static org.apache.commons.lang.exception.ExceptionUtils.getRootCauseMessage;
+import static org.xwiki.attachment.internal.AttachmentsListener.HINT;
+
+/**
+ * Listen for attachments creation and if the created attachment name matches a redirection, remove the corresponding
+ * XObject.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Singleton
+@Named(HINT)
+public class AttachmentsListener implements EventListener
+{
+    /**
+     * The hint for this class.
+     */
+    public static final String HINT = "Attachment.Code.RedirectAttachmentClass";
```

Comment:
Sounds like a hint that shouldn't be change because it's attached to some kind of mechanism? Would be good to document then. I use to use the same hints as the FQN of the class for listeners usually personally.

Reply:
I'm not sure of the documentation you are missing here :)  I'll change the hint to the fqn.

Reply:
> I'm not sure of the documentation you are missing here :)

it was in case the hint was derived from a mechanism involving this class but apparently it's not the case.